Marlin/src/lcd/menu/menu_probe_offset.cpp

```diff
@@ -99,30 +99,22 @@ void probe_offset_wizard_menu() {
     do_z_clearance(0
     #ifdef Z_AFTER_HOMING
       +Z_AFTER_HOMING
-    #elseif defined(Z_HOMING_HEIGHT)
+    #elif defined(Z_HOMING_HEIGHT)
       +Z_HOMING_HEIGHT
     #endif
     );
   });
 
   ACTION_ITEM(MSG_BUTTON_CANCEL, []{
     set_offset_and_go_back(z_offset_backup);
-
     // If wizard-homing was done with PROBE_OFFSET_WIZARD_START_Z by probe.
-    // Set Z to 0 (if higher than 0) to be sure for G28 clearance and rehome Z with backed up offset.
-    #if defined(PROBE_OFFSET_WIZARD_START_Z) && EITHER(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN, USE_PROBE_FOR_Z_HOMING)
-      if(current_position.z > 0) { current_position.z = 0; sync_plan_position();; }
```

Comment:
I realy think we need this. Because if the Nozzle is close to the bed but the current_position.z is too high, there may will not be enough clearance for probing!
`if(current_position.z > 0) { current_position.z = 0; sync_plan_position(); }`

Reply:
I believe the better solution is to use `set_axis_never_homed(Z_AXIS)` and there is already precedent for that.